src/accelerate/utils/memory.py

```diff
@@ -103,7 +103,7 @@ def should_reduce_batch_size(exception: Exception) -> bool:
     return False
 
 
-def find_executable_batch_size(function: callable = None, starting_batch_size: int = 128):
+def find_executable_batch_size(function: callable = None, starting_batch_size: int = 128, reduce_batch_size_fn: callable = None):
     """
     A basic decorator that will try to execute `function`. If it fails from exceptions related to out-of-memory or
     CUDNN, the batch size is cut in half and passed to `function`
```

Comment:
Let's make sure to add `reduce_batch_size_fn` to the docstring, describing what it should take in and return (a batch size and return a modified batch size?)